csvcubed/csvcubed/models/cube/qb/components/codelist.py

```diff
@@ -159,8 +159,17 @@ def get_permitted_rdf_fragment_hints(self) -> Set[RdfSerialisationHint]:
     def get_default_node_serialisation_hint(self) -> RdfSerialisationHint:
         return RdfSerialisationHint.ConceptScheme
 
-    def validate_data(self, data: PandasDataTypes) -> List[ValidationError]:
-        return []  # TODO: implement this.
+    def validate_data(self, data: PandasDataTypes) -> list[Optional[ValidationError]]:
+        """
+        Validate the data held in the codelists, assuming case insensitivity
+        """
+
+        errors = list()
+
+        if isinstance(data, Series):
```

Comment:
I'd be tempted to do something which is guaranteed to work with all `PandasDataTypes` - perhaps converting the other types to a Series?

Reply:
Okay.